From 445de105c0096385c88bf79946cb15b9e3a0d40a Mon Sep 17 00:00:00 2001
From: Johannes Agricola <johannes.agricola@work-microwave.com>
Date: Fri, 27 Aug 2021 11:17:18 +0200
Subject: [PATCH 11/14] This patch reduces the possible press-interval for the
 front-panel keys be introducing a high (50ms instead of 5ms) debounce default
 interval. This patch is only needed if the device tree has no
 debounce-interval set, e.g.

		rightkey {
			gpios = <&gpio24 2 1>;
			interrupts = <8 0x2 0 0>;
			linux,code = <106>;		/* KEY_RIGHT */
			debounce-interval = <50>;
		};

Without this patch very frequent errors and wrong key-presses happen, e.g. when
pressing only RIGHT key:

[   72.621202] fpctrl <inf> RIGHT
[   72.810853] fpctrl <inf> RIGHT
[   72.967489] fpctrl <inf> RIGHT
[   73.968776] pca953x 1-0024: failed reading register 2: -110
[   73.995423] fpctrl <inf> RIGHT
[   74.123525] fpctrl <inf> RIGHT
[   74.235027] fpctrl <inf> TXON
[   74.243534] satctl <inf> Turning TX on.
[   74.251413] fpctrl <inf> RIGHT
[   74.365604] fpctrl <inf> RIGHT
[   74.645672] satctl <inf> Demodulator locked.
[   74.655464] fpctrl <inf> RIGHT
[   74.832994] satctl <inf> Demodulator had a very short unlocked state.
[   74.963224] fpctrl <inf> RIGHT
[   75.131921] fpctrl <inf> RIGHT
[   75.180064] satctl <inf> Demodulator locked.
[   75.297473] fpctrl <inf> RIGHT
[   75.434722] fpctrl <inf> RIGHT

Errors come from pca953x_read_single() in gpio-pca953x.c and is "Connection
timed out" and "Input/output error".
---
 drivers/input/keyboard/gpio_keys.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/keyboard/gpio_keys.c b/drivers/input/keyboard/gpio_keys.c
index a5dc4ab87fa1..5ea18bc2b4db 100644
--- a/drivers/input/keyboard/gpio_keys.c
+++ b/drivers/input/keyboard/gpio_keys.c
@@ -783,7 +783,7 @@ gpio_keys_get_devtree_pdata(struct device *dev)
 
 		if (fwnode_property_read_u32(child, "debounce-interval",
 					 &button->debounce_interval))
-			button->debounce_interval = 5;
+			button->debounce_interval = 50;
 
 		button++;
 	}
-- 
2.25.1

